httpclient/metrics_round_tripper.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright © 2024 Acronis International GmbH.
+
+Released under MIT license.
+*/
+
+package httpclient
+
+import (
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+var (
+	ClientHTTPRequestDuration *prometheus.HistogramVec
+	ClassifyRequest           func(r *http.Request, reqType string) string
+)
+
+// MustInitAndRegisterMetrics initializes and registers external HTTP request duration metric.
+// Panic will be raised in case of error.
+func MustInitAndRegisterMetrics(namespace string) {
```

Comment:
Developers need to call this function explicitly, but it’s not immediately clear where or when to do so. It’s easy to imagine a scenario where they forget (a likely case), construct a client using the httpclient.New() method, and expect metrics to be collected, which they won’t be. This makes the current approach error-prone.

A better approach would be to pass the metrics collector explicitly. For reference, you can see a similar implementation in the middleware here: [httpserver/middleware/metrics.go](https://github.com/acronis/go-appkit/blob/master/httpserver/middleware/metrics.go).

Additionally, accepting an interface instead of a concrete object would provide greater flexibility, allowing developers to use custom implementations, such as a non-Prometheus metrics collector.